src/Management/src/Endpoint/Actuators/HttpExchanges/HttpExchangesDiagnosticObserver.cs

```diff
@@ -0,0 +1,217 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Steeltoe.Management.Diagnostics;
+
+namespace Steeltoe.Management.Endpoint.Actuators.HttpExchanges;
+
+internal sealed class HttpExchangesDiagnosticObserver : DiagnosticObserver, IHttpExchangesRepository
+{
+    private const string DiagnosticName = "Microsoft.AspNetCore";
+    private const string DefaultObserverName = "TraceDiagnosticObserver";
+    private const string StopEventName = "Microsoft.AspNetCore.Hosting.HttpRequestIn.Stop";
+
+    internal const string Redacted = "******";
+
+    private static readonly DateTime BaseTime = DateTime.UnixEpoch;
+    private readonly IOptionsMonitor<HttpExchangesEndpointOptions> _optionsMonitor;
+    private readonly ILogger<HttpExchangesDiagnosticObserver> _logger;
+    internal ConcurrentQueue<HttpExchange> Queue { get; } = new();
+
+    public HttpExchangesDiagnosticObserver(IOptionsMonitor<HttpExchangesEndpointOptions> optionsMonitor, ILoggerFactory loggerFactory)
+        : base(DefaultObserverName, DiagnosticName, loggerFactory)
+    {
+        ArgumentNullException.ThrowIfNull(optionsMonitor);
+
+        _optionsMonitor = optionsMonitor;
+        _logger = loggerFactory.CreateLogger<HttpExchangesDiagnosticObserver>();
+    }
+
+    public HttpExchangesResult GetHttpExchanges()
+    {
+        HttpExchange[] recentExchanges = Queue.ToArray();
+        foreach (HttpExchange exchange in recentExchanges)
+        {
+            FilterHeaders(exchange.Request.Headers, _optionsMonitor.CurrentValue.RequestHeaders);
+            FilterHeaders(exchange.Response.Headers, _optionsMonitor.CurrentValue.ResponseHeaders);
+        }
+
+        return new HttpExchangesResult(recentExchanges);
+    }
+
+    public override void ProcessEvent(string eventName, object? value)
```

Comment:
I think we should drop `HttpClientDesktopObserver` (which reads from `WebRequest`) entirely, because it was [obsoleted in .NET 6](https://learn.microsoft.com/en-us/dotnet/core/compatibility/networking/6.0/webrequest-deprecated).

Then rename the renaming one to just `HttpClientObserver`.

Reply:
`value` is an anonymous type with a `HttpContext` as a property... I can't seem to get type checking to work in this scenario and haven't found anything in several web searches.

You're probably right about the other observers, I'm not sure yet if it's something I want to include in this PR though

Reply:
> value is an anonymous type with a HttpContext as a property... I can't seem to get type checking to work in this scenario and haven't found anything in several web searches.

These are just the tests acting silly by passing an anonymous type. As can be seen [here](https://github.com/dotnet/aspnetcore/blob/main/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L88), ASP.NET always passes in a `HttpContext` instance. So the code can just be:
```c#
if (value is HttpContext httpContext)
{
    RecordHttpExchange(current, httpContext);
}
```
When using that and debugging the test `AddSeveralActuators_WebApplicationBuilder_NoConflict` (breakpoint in method `HttpExchangesDiagnosticObserver.ProcessEvent`), it always sends an `HttpContext` instance.

> You're probably right about the other observers, I'm not sure yet if it's something I want to include in this PR though

Why not, do you have any concerns? The current names are just confusing.

Reply:
> These are just the tests acting silly by passing an anonymous type...

I should know by now not to trust that these tests are always doing the right thing 🙄 

> Why not, do you have any concerns? The current names are just confusing. The other observers suffer from the same issues as addressed in this PR, such as obtaining `Activity.Current` when not needed. Please open an issue otherwise.

Not any specific concerns, more a question of maintaining focus... If I have time to tackle it before your next review I will, otherwise I'll create a follow-up issue so this PR doesn't keep dragging on

Reply:
I can do the delete & rename, but it looks like we can't currently adjust the reflection-based property access code: https://github.com/dotnet/runtime/issues/82910

Reply:
Oh right... I didn't come back and check on other usages of `GetHttpContextPropertyValue` after trying to adjust the other observers. I'll remove this method and the two tests. The runtime issue is the reason why I can't do the same for the HttpClient Observer(s)

Reply:
Okay, I understand that `HttpClientObserver` needs to keep using reflection due to the runtime issue. But that doesn't apply to `AspNetCoreHostingObserver`, so please update to remove the reflection there.